### DISCUSSION TITLE
integers of different sizes

### DISCUSSION THREAD

**Discussion by RavikumarTulugu** [taocpp/PEGTL#209] [Q&A] [answered]
i am looking for support for representing integers of different sizes like int8 , int16, int32 , int64 , i don't see any support readily available in the library. please advise.

**Comment by ColinH** [comment]
Have you already checked `include/tao/pegtl/contrib/integer.hpp`?

**Comment by RavikumarTulugu** [comment]
yeah, but pardon my C++ ignorance, i looked but still clueless on the usage. any example would be of great help. my requirement is i have sized integers like int8 , int16 , int32 , int64 in my example and i am clueless on how to define them.

**Comment by ColinH** [comment]
Could you please elaborate on how exactly your question is connected to parsing? What are you trying to achieve with your example? Parse integers?

**Comment by RavikumarTulugu** [comment]
yes, while parsing integers how do we enforce the limits for different sizes ??

**Comment by d-frey** [comment, marked as answer]
You can use `maximum_rule< std::uint8_t >`, as seen in `tao/pegtl/contrib/uri.hpp`. The maximum_rule allows you to specify the datatype into which a parsed number must fit, as well as an optional maximum value. You could use `maximum_rule< std::uint8_t, 123 >` as a rule that parses/matches any number from 0 to 123.
